deployment/keystone/deploy.go

```diff
@@ -421,8 +420,36 @@ type RegisterNOPSResponse struct {
 	Nops []*kcr.CapabilitiesRegistryNodeOperatorAdded
 }
 
-func RegisterNOPS(ctx context.Context, req RegisterNOPSRequest) (*RegisterNOPSResponse, error) {
-	nops := req.Nops
+func RegisterNOPS(ctx context.Context, lggr logger.Logger, req RegisterNOPSRequest) (*RegisterNOPSResponse, error) {
+	existingNops, err := req.Registry.GetNodeOperators(&bind.CallOpts{})
+	if err != nil {
+		return nil, err
+	}
+	existingNopsAddrToID := make(map[string]uint32)
+	for id, nop := range existingNops {
+		existingNopsAddrToID[nop.Admin.String()] = uint32(id)
```

Comment:
the admin strings are phony in some of our deployments and not unique.

maybe hash the name,addr and use that as the key

Reply:
i think you can just use the struct as the key